__test__/integrationTests.test.ts

```diff
@@ -285,3 +285,58 @@ describe("Integration tests for AUTH routes:", () => {
         });
     });
 });
+
+// Post route handler integration tests
+// -------------------------------------------------------------------------
+describe("Integration tests for POST routes:", () => {
+    describe("/api/posts/create", () => {
+        test("responds with 201 Created and returns the created post", async () => {
+            // Create a user first
+            await axios.post("/api/auth/register", {
+                userName: "postUser",
+                email: "post@email.com",
+                password: "Abc-1234",
+            });
+            expect(await db.user.count()).toBe(1);
+
+            // Log the user in
+            const loginResponse = await axios.post("/api/auth/login", {
+                email: "post@email.com",
+                password: "Abc-1234",
+            });
+            expect(loginResponse.status).toBe(200);
+
+            // Get session cookie
+            if (!loginResponse.headers["set-cookie"])
+                throw new Error("No cookie set");
+            const cookie: string = loginResponse.headers["set-cookie"][0];
+
+            const response = await axios({
+                url: "/api/posts/create",
+                method: "POST",
+                headers: {
+                    Cookie: cookie,
+                },
+                data: {
+                    title: "test",
+                    content: "test",
+                    status: "DRAFT",
+                    category: "GENERAL",
```

Comment:
should the tags be included in the data?

Reply:
Yes, you are right. I'll add that on Monday!

Reply:
Tags are now inserted when a post is created. Please be aware that the way it is currently set up requires us to create a list of possible tags in the database. Only tags already in the database can be selected. We should consider some way to extend this list as people actually use the site at a future date.